packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -61,15 +61,19 @@ export class BlockList extends Component {
 		// create an empty block of the selected type
 		const newBlock = createBlock( itemValue );
 
-		this.finishBlockAppendingOrReplacing( newBlock );
+		this.finishInsertingOrReplacingBlock( newBlock );
 	}
 
-	finishBlockAppendingOrReplacing( newBlock ) {
-		// now determine whether we need to replace the currently selected block (if it's empty)
-		// or just add a new block as usual
+	finishInsertingOrReplacingBlock( newBlock ) {
 		if ( this.isReplaceable( this.props.selectedBlock ) ) {
-			// do replace here
+			// replace selected block
 			this.props.replaceBlock( this.props.selectedBlockClientId, newBlock );
+		} else if ( this.props.isPostTitleSelected && this.isReplaceable( this.props.firstBlock ) ) {
+			// replace first block in post: there is no selected block when the post title is selected,
+			// so replaceBlock does not select the new block and we need to manually select the new block
+			const { clientId: firstBlockId } = this.props.firstBlock;
+			this.props.replaceBlock( firstBlockId, newBlock );
+			this.props.selectBlock( newBlock.clientId );
```

Comment:
Some additional context for this line:

If _replacing_ the first block of the post when the title selected, it is also necessary to select the new block at the beginning of the post. This is because the block selection reducer checks and only [updates the block selection](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/reducer.js#L1003) if [the action's clientId matches the clientId from that state's `start` property](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/reducer.js#L998). When the post's title is selected, the state's `start` property is an empty object, so the undefined `clientId` does not match the action's `clientId` and the block selection is not updated. When a block is selected (such as if you're replacing an empty paragraph block) then the `clientId`'s do match and the selection state is updated, resulting in the new block being selected. I feel that just sending of the `SELECT_BLOCK` action works well for this rather-edge case, and it avoids the complexity/risk inherent in updating the `blockSelection` reducer we share with web. I'm certainly open to suggestions for better ways to handle this though.